test/unit/kubernetes-deploy/resource_watcher_test.rb

```diff
@@ -33,38 +33,67 @@ def test_failure_with_mock_resource
   end
 
   def test_timeout_from_resource
-    resource = build_mock_resource("timeout")
+    resource = build_mock_resource(final_status: "timeout")
 
-    watcher = KubernetesDeploy::ResourceWatcher.new([resource])
+    watcher = KubernetesDeploy::ResourceWatcher.new([resource], logger: test_logger)
     watcher.run(delay_sync: 0.1)
 
     assert_logs_match(/Waiting for web-pod with 1s timeout/)
     assert_logs_match(/web-pod failed to deploy with status 'timeout'/)
     assert_logs_match(/Spent (\S+)s waiting for web-pod/)
   end
 
+  def test_wait_logging_when_resources_do_not_finish_together
```

Comment:
The test class refactors are for the purposes of this test, which will become more meaningful with the upcoming logging changes.